### PR TITLE
Queue, File and Datalake Sas

### PR DESCRIPTION
**Pull request by gapra-msft**

### CODE REVIEW

**Comment by gapra-msft** [comment]
TODO : Add notes to the generateSAS methods that the client must be authenticated via StorageSharedKeyCredential

**Comment by gapra-msft** [comment]
TODO: https://github.com/Azure/azure-sdk-for-java/issues/6713

File: sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/APISpec.groovy [hunk 1]
```diff
@@ -43,7 +43,7 @@ class APISpec extends Specification {
     String connectionString
 
     // If debugging is enabled, recordings cannot run as there can only be one proxy at a time.
-    static boolean enableDebugging = false
+    static boolean enableDebugging = true
```
Comment (rickle-msft) [inline sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/APISpec.groovy]: revert

Comment (rickle-msft) [reply sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/APISpec.groovy]: Should actually probably be removed in favor of the environment variable a la blobs.

**Pull request merged**